lib/chef/provider/windows_task.rb

```diff
@@ -214,6 +217,38 @@ def task_need_update?
         false
       end
 
+      def start_day_updated?
+        current_day = convert_system_date_format_into_ruby_date(@current_resource.start_day)
+        new_day = convert_system_date_format_into_ruby_date(new_resource.start_day)
+        current_day != new_day
+      end
+
+      def start_time_updated?
+        time = convert_system_date_format_into_ruby_date(@current_resource.start_time).strftime("%H:%M")
+        time != new_resource.start_time
+      end
+
+      def convert_system_date_format_into_ruby_date(date_in_string)
+        Chef::Log.debug "Looking system date format"
+        task_script = <<-EOH
+          [Console]::OutputEncoding = [Text.UTF8Encoding]::UTF8
+          ([datetime]'#{date_in_string}').ToString('yyyy-MM-dd HH:mm')
+        EOH
+        # Convert given date time in string into `yyyy-MM-dd HH:mm` formt
+        date_time = powershell_out(task_script).stdout.force_encoding("UTF-8").gsub(/[\s+\uFEFF]/, "")
+        # Convert `yyyy-MM-dd HH:mm` formated string date into ruby DateTime object
+        DateTime.strptime(date_time, "%Y-%m-%d %H:%M")
+      end
+
+      def convert_user_date_to_system_date
+        task_script = <<-EOH
+          [Console]::OutputEncoding = [Text.UTF8Encoding]::UTF8
+          ([datetime]"#{new_resource.start_day}").ToString([Globalization.Cultureinfo]::CurrentCulture.DateTimeFormat.ShortDatePattern)
+        EOH
+        # Convert start_date into OS date format
+        powershell_out(task_script).stdout.force_encoding("UTF-8").gsub(/[\s+\uFEFF]/, "")
```

Comment:
Do we need to shell out like this to know what the system date format is? Could we do without the shelling out? It's slow.

Why can't we do something like this:
```
 DateTime.parse('8/3/2017 15:12').strftime("%Y-%m-%d %H:%M")
```

Reply:
@btm The `start_day`  to execute the command, is required in system date format and if two nodes are in diff system DateTime format then the resource execution will fail so need to convert system date format.
like if node A date format in mm-dd-yyyy and node B date format in yyyy-mm-dd and user has passed date like 11-11-2015 then for node B  the resource will fail to execute.
and the command execution needs system date format so we can't do like 
```
DateTime.parse('8/3/2017 15:12').strftime("%Y-%m-%d %H:%M")
```

Reply:
Do you have access to a system where the system format is different? What do these two methods return on that system in Ruby?

```
Time.new.strftime('%x %X')
Time.new.strftime('%D %T')
```

Reply:
agree with @btm. It would be preferable to do the date comparisons in ruby

Reply:
@btm , @mwrock I have removed `convert_system_date_format_into_ruby_date` function and used `DateTime.parse` to convert string into ruby date object  
https://github.com/chef/chef/pull/6312/commits/1a8f50ac05609e7bbbb3dd70c583d11bc7414753

But to execute window_task command(schtasks) `start_day` must be in the system date format. By default windows system date is in `MM/dd/yyyy` format, but the user can change it to any other format e.g. yyyy-MM-dd or dd-MM-yyyy, etc. So we need to convert `start_day` into system date format

The Time.new.strftime('%x %X') Time.new.strftime('%D %T') gives date into `MM/dd/yy` format it's output does not depend on system date format

```
irb(main):001:0> Time.new.strftime('%x %X')
=> "08/30/17 18:51:06"
irb(main):002:0> Time.new.strftime('%D %T')
=> "08/30/17 19:00:27"
```

*Another Solution:* We can restrict user to give start_day as input to windows_task resource into  system date format 
*Problem:* If node A has default date format(M/d/yyyy) and for node B user had changed to d-M-yy and the user has passed date in dafault date format(M/d/yyyy) from recipe then the execution fails for node B because start_day required in system date format.

Reply:
Forcing the user to use the system format could prove overly restrictive and break some resources. While doing this in ruby would be ideal, resorting to shelling out is probably acceptable.